src/dswx_sar/region_growing.py

```diff
@@ -0,0 +1,308 @@
+import time
+import os
+import numpy as np
+from scipy import ndimage
+import cv2
+
+import time
+import logging
+import mimetypes
+from joblib import Parallel, delayed
+
+from dswx_sar import dswx_sar_util
+from dswx_sar.dswx_runconfig import RunConfig, _get_parser
+
+
+logger = logging.getLogger('dswx_s1')
+
+
+def region_growing(fuzz_image,
+                   initial_seed=0.6,
+                   tolerance=0.45,
+                   maxiter=200):
+    """The regions are then grown from the seed points to adjacent points
+    since it covers the tolerance values.
+
+    Parameters
+    ----------
+    fuzz_image : numpy.ndarray
+        fuzzy image with values [0, 1] where 0 is non-water and 1 is water
```

Comment:
Good suggestions. But, region-growing is used in different ways in initial thresholding, regardless of water. So more general name seems better.

Reply:
How do you like `likelihood_image`? It's more specific than `fuzz` but still general since `water` is not in the variable name. Or is my interpretation of values as [0-1] as a likelihood incorrect or too narrow?

Also for clarification about the comment above, do you mean this region-growing code can/will be used for features besides water? If so, would it make sense remove `water` from variables names?

Sorry for all these questions...